add github pull request template (#2894)
GoogleContainerTools/skaffold: .github/pull_request_template.md

<!-- 🎉🎉🎉 Thank you for the PR!!! 🎉🎉🎉 -->

Relates to _in case of new feature, this should point to issue/(s) which describes the feature_

> Fixes _in case of a bug fix, this should point to a bug and any other related issue(s)_

Reviewer: ```suggestion
Fixes #<issue number>. _in case of a bug fix, this should point to a bug and any other related issue(s)_
```

> Should merge before : _list any dependent PRs_

Reviewer: ```suggestion
Should merge before : _list any PRs that depend on this PR_
Should merge after : _list any PRs that are prerequisites to this PR_
```

# Description

<!-- Describe your changes here- ideally you can get that description straight from
your descriptive commit message(s)! -->

> # Output Changes

Reviewer: ```suggestion
# User facing changes
```

> Write n/a if not output or log lines added

Reviewer: ```suggestion
Write n/a if not output or log lines changed and no behavior is changed
```

## Before

> Paste skaffold output before your change

Reviewer: ```suggestion
If log/output changes: Paste the current relevant skaffold output 
If behavior changes: describe succintly the current behavior
```

## After

> Paste skaffold output after your change

Reviewer: ```suggestion
If log/output changes: Paste skaffold output after your change
If behavior changes: describe succintly the behavior after your change
```

## Next PRs.

In this section describe a list of follow up PRs if the current PR is a part of big feature change.

See example #2811

Write n/a if not applicable.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [unit tests](../DEVELOPMENT.md#creating-a-pr)

> - [ ] Mentions any output changes. 

Reviewer: ```suggestion
- [ ] Mentions any output changes. 
- [ ] Adds integration tests if needed
```

_See [the contribution guide](../CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a example to the `examples` dir](https://github.com/GoogleContainerTools/skaffold/tree/master/examples), please copy them to [`integration/examples`](https://github.com/GoogleContainerTools/skaffold/tree/master/integration/examples)
- Every new example added in [`integration/examples` dir](https://github.com/GoogleContainerTools/skaffold/tree/master/integration/examples), should be tested in [integration test](https://github.com/GoogleContainerTools/skaffold/tree/master/integration)

## Reviewer Notes

- [ ] The code flow looks good. 
- [ ] Unit test added.

> - [ ] User facing document changes look good.

Reviewer: ```suggestion
- [ ] User facing changes look good.
```
